internal/validitywindow/validitywindow_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validitywindow
+
+import (
+	"context"
+	"encoding/binary"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidityWindowVerifyExpiryReplayProtection(t *testing.T) {
+	tests := []struct {
+		Name          string
+		Accepted      []executionBlock
+		VerifyBlock   executionBlock
+		OldestAllowed int64
+		ExpectedError error
```

Comment:
Can we unexport the field names here?

Reply:
Can we add an `Accepted` field to indicate the last block considered accepted as opposed to in processing as we do for `TestValidityWindowIsRepeat` below?

Reply:
not sure exactly how to do that, since we really need the included chunks to be set correctly, as it's being tested.

Reply:
Why can't we do the exact same as we do below and just test `VerifyExpiryReplayProtection` instead of `IsRepeat` ?

Reply:
```golang
			r := require.New(t)

			chainIndex := &testChainIndex{}
			tvw := NewTimeValidityWindow(&logging.NoLog{}, trace.Noop, chainIndex)
			r.NotNil(tvw)
			for i, blk := range test.blocks {
				if i <= int(test.accepted) {
					tvw.Accept(blk)
				}
				chainIndex.set(blk.GetID(), blk)
			}
```

Reply:
oh - now I understood what you meant. You want that the last entry in the blocks would be the one used for the `VerifyExpiryReplayProtection`